### PR TITLE
aarch64: conv: injector for 1x1 conv

### PR DESCRIPTION
**Pull request by kawakami-k**
# Description

This PR supports injectors for 1x1 convolution. 
The following PRs may be required before merge this PR.
- https://github.com/oneapi-src/oneDNN/pull/1556 
- https://github.com/oneapi-src/oneDNN/pull/1557 
- https://github.com/oneapi-src/oneDNN/pull/1558 
- https://github.com/oneapi-src/oneDNN/pull/1559

Thank you.

# Checklist

## General

- [x] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
- [x] Have you formatted the code using clang-format?

### CODE REVIEW

File: src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_version.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/*******************************************************************************
+ * Copyright 2022-2023 FUJITSU LIMITED
```
Comment (igorsafo) [inline src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_version.h:2]: Just to double check, is it really 2020-2023 or should it be 2023?

**Review by igorsafo — APPROVED** [review]

**Comment by densamoilov** [comment]
Hi @nSircombe, may we ask you to test the following PRs from Fujitsu to make sure they don't break anything (public CI doesn't work):
* #1556 
* #1557 
* #1558 
* #1559 
* #1561

**Comment by nSircombe** [comment]
Hi @densamoilov,
Sure, I'll add them to a downstream branch and try.
One problem that has just appeared though, which I think we will need to look at first, is a test failure in wino which has snuck in between https://github.com/oneapi-src/oneDNN/pull/1555 going up and being merged, during which time our builds of the upstream master were red.

**Comment by annop-w** [comment]
The wino test failure that @nSircombe mentioned actually shows up only for oneDNN Threadpool build. I can reproduce the failure with oneDNN v2.7. This error slipped through our CI because of a bug in our CI script for the threadpool build. So, the wino failure is not a blocker for these chain of commits.

**Comment by nSircombe** [comment]
Hi @densamoilov,
I've tried the union of those patches on our build pipelines. The only one that passes is the one on Neoverse-V1 - i.e. SVE hardware. For the equivalent run on N1 (non SVE) I'm seeing failures of the form:

```
96% tests passed, 5 tests failed out of 122

Total Test time (real) = 1291.07 sec

The following tests FAILED:
	  1 - cpu-bnorm-u8-via-binary-postops-cpp (Failed)
	 40 - test_binary (Failed)
	 96 - test_benchdnn_binary_ci_cpu (Failed)
	 97 - test_benchdnn_binary_different_dt_ci_cpu (Failed)
	111 - test_benchdnn_reorder_ci_cpu (ILLEGAL)
Errors while running CTest
```

I will re-run to double check.

**Comment by kawakami-k** [comment]
@nSircombe 
Sorry, I forgot to run CI on no-SVE machine... I'll try it. 

Recently, the automatic CI for Arm environment has stopped working, but I hope there is an alternative...

**Comment by vpirogov** [comment]
Travis CI supports Arm, but it's not free...

**Comment by kawakami-k** [comment]
@nSircombe 

I updated
https://github.com/oneapi-src/oneDNN/pull/1559

The difference can be seen in
https://github.com/oneapi-src/oneDNN/compare/e544bf5957b972828231a8a4cfeb366755aa5116..01eac13f6b2bda143ab852884108cd45ebac7197

Could you check if this patch will work for no-SVE machine?

**Comment by nSircombe** [comment]
Hi @kawakami-k, 
I've applied that [change](https://github.com/oneapi-src/oneDNN/compare/e544bf5957b972828231a8a4cfeb366755aa5116..01eac13f6b2bda143ab852884108cd45ebac7197) but I still see the illegal instructions in `test_benchdnn_reorder_ci_cpu`:

```
The following tests FAILED:
	111 - test_benchdnn_reorder_ci_cpu (ILLEGAL)
Errors while running CTest
```

**Comment by kawakami-k** [comment]
@nSircombe 

Thank you for the trial. I also updated the reorder.
Could you please use 
https://github.com/kawakami-k/oneDNN/commit/b7d210bb626610a4e924fcee3b143c81eeb68bc1

The following commit log is a summary of the pull requests I have submitted.
https://github.com/kawakami-k/oneDNN/commits/pr_all

However, I need to resolve an error in `tests/gtests/test_iface_wino_convolution`.
https://github.com/oneapi-src/oneDNN/issues/1577

**Comment by nSircombe** [comment]
Hi @kawakami-k,

Running https://github.com/kawakami-k/oneDNN/commit/b7d210bb626610a4e924fcee3b143c81eeb68bc1 I'm now seeing a failure in `test_graph_unit` - a new test on `master` I believe.
This is only for V1 (SVE-256), the N1 (Neon) builds are OK.
Our nightly build of the upstream master shows this test as passing on both N1 and V1 - have you seen this failure on A64FX?

**Comment by nSircombe** [comment]
Building with ` -DONEDNN_BUILD_GRAPH=0 -DDNNL_ENABLE_JIT_PROFILING=0`, https://github.com/kawakami-k/oneDNN/commit/b7d210bb626610a4e924fcee3b143c81eeb68bc1 passes all our tests OK.

We've raised an issue for build failures we've seen with the JIT profiler (hence -DDNNL_ENABLE_JIT_PROFILING=0) - https://github.com/oneapi-src/oneDNN/issues/1582

**Comment by dzarukin** [comment]
Hi @kawakami-k, could you, please, rebase changes on latest master? Thanks!

File: src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_version.h (hunk 1 above)
Comment (kawakami-k) [reply src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_version.h:2]: This file is coped from 
https://github.com/fujitsu/xbyak_aarch64/commits/main/xbyak_aarch64/xbyak_aarch64_version.h
, which was firstly disclosed in Nov. 2022.

**Comment by kawakami-k** [comment]
@dzarukin 
I rebased the changes on the latest master.

Thank you!

**Comment by igorsafo** [comment]
@kawakami-k The changes landed into `master`, thanks for the contribution!

**Pull request merged**